Mono.Addins.Setup/Mono.Addins.Setup/Repository.cs

```diff
@@ -128,21 +128,27 @@ public IAsyncResult BeginDownloadSupportFile (string name, AsyncCallback cb, obj
 			}
 
 			res.FilePath = cachedFile;
-			WebRequestHelper.GetResponseAsync (() => (HttpWebRequest)WebRequest.Create (u)).ContinueWith (t => {
+			var request = DownloadFileRequest.DownloadFile (u.ToString (), false).ContinueWith (t => {
 				try {
-					var resp = t.Result;
-					string dir = Path.GetDirectoryName (res.FilePath);
-					lock (this) {
-						if (!Directory.Exists (dir))
-							Directory.CreateDirectory (dir);
-					}
-					byte[] buffer = new byte [8092];
-					using (var s = resp.GetResponseStream ()) {
-						using (var f = File.OpenWrite (res.FilePath)) {
-							int nr = 0;
-							while ((nr = s.Read (buffer, 0, buffer.Length)) > 0)
-								f.Write (buffer, 0, nr);
+					using (var resp = t.Result) {
+						string dir = Path.GetDirectoryName (res.FilePath);
+						lock (this) {
```

Comment:
Why lock here?

Reply:
This is the original code. I have not changed this.